spec/services/container_dashboard_service_spec.rb

```diff
@@ -0,0 +1,53 @@
+require "spec_helper"
+
+describe ContainerDashboardService do
+  before(:each) do
+    ManageIQ::Providers::ContainerManager.delete_all
+  end
+
+  context "providers" do
+    it "returns all providers correctly" do
+      Tempfile.create(['config', 'yml']) do |f|
+        f.write Psych.dump(['ems-type:openshift',
+                            'ems-type:kubernetes',
+                            'ems-type:atomic',
+                            'ems-type:openshift_enterprise',
+                            'ems-type:atomic_enterprise'])
+        f.close
+
+        Vmdb::PermissionStores.configure do |config|
+          config.backend = 'yaml'
+          config.options[:filename] = f.path
+        end
+
+        Vmdb::PermissionStores.initialize!
+      end
+
+      FactoryGirl.create(:ems_kubernetes, :hostname => "test1.com")
+      FactoryGirl.create(:ems_openshift, :hostname => "test2.com")
+      FactoryGirl.create(:ems_openshift_enterprise, :hostname => "test3.com")
+      FactoryGirl.create(:ems_atomic, :hostname => "test4.com")
+      FactoryGirl.create(:ems_atomic_enterprise, :hostname => "test5.com")
+
+      providers_data = ContainerDashboardService.new(nil, nil).providers
+      expect(providers_data).to eq([{
+                                      :iconClass    => "pficon pficon-kubernetes",
+                                      :count        => 1,
+                                      :id           => :kubernetes,
+                                      :providerType => :Kubernetes
+                                    },
+                                    {
+                                      :iconClass    => "pficon pficon-openshift",
+                                      :count        => 2,
+                                      :id           => :openshift,
+                                      :providerType => :Openshift
+                                    },
+                                    {
+                                      :iconClass    => "pficon pficon-atomic",
+                                      :count        => 2,
+                                      :id           => :atomic,
+                                      :providerType => :Atomic
+                                    }])
```

Comment:
> How is this test checking the change in this PR? (or in other words - what is this test supposed to test?)

Its more of a generic test to see that the method works correctly. Now that I think of it, it should check for filtering of providers with zero entity count.